app/models/package_manager/go.rb

```diff
@@ -161,15 +165,15 @@ def self.mapping(raw_project)
           # if this is a versioned module, make sure to find the right versioned project
           if versioned_module_regex
             # try and find a versioned name matching this repository_url
-            existing_project_name = Project.where(platform: "Go").where("lower(repository_url) = :repo_url and name like :name", repo_url: url.downcase, name: "%/#{versioned_module_regex[2]}").first&.name
+            existing_project_name = Project.visible.where(platform: "Go").where("lower(repository_url) = :repo_url and name like :name", repo_url: url.downcase, name: "%/#{versioned_module_regex[2]}").first&.name
 
             # if we didn't find one then try and get the base project
             unless existing_project_name.present? # rubocop: disable Metrics/BlockNesting
-              versioned_name = Project.where(platform: "Go").where("lower(repository_url) = ? and name not like '%/v'", url.downcase).first&.name
+              versioned_name = Project.visible.where(platform: "Go").where("lower(repository_url) = ? and name not like '%/v'", url.downcase).first&.name
               existing_project_name = versioned_name&.concat("/#{versioned_module_regex[2]}")
             end
           else
-            existing_project_name = Project.where(platform: "Go").where("lower(name) = ?", raw_project[:name].downcase).first&.name
+            existing_project_name = Project.visible.where(platform: "Go").lower_name(raw_project[:name].downcase).first&.name
```

Comment:
seems like we could assume by this point that we have the correct name now, so might not need the `lower()` anymore. WDYT?

Reply:
I think we will still run into problems if the project is not a module project. The proxy will echo back whatever string you sent. We don't have too many duplicate names right now, a little less than 2k, so I might be able to figure out which one is "right" and remove the "wrong" one so this code would only ever find one and it would end up being the correct name. I could also look into figuring out if the proxy responded with the correct name and pass that as a boolean through to this where it can better determine how the name should be set.

Reply:
After re-reading this section a bit, this name matching code is only for non module names that we get. Since those can't be considered canonical due to the way the proxy works we will still need to keep this matching. I've added a spec to capture that use case. I've also checked duplicate names again and cleaned up that data.